code/modules/mob/living/carbon/life.dm

```diff
@@ -235,6 +241,43 @@
 	if(reagents)
 		reagents.metabolize(src)
 
+/mob/living/carbon/handle_medical_effects()
+	for(var/datum/medical_effect/E in medical_effects)
+		E.process(src)
+
+/mob/living/carbon/proc/add_medical_effect(var/datum/medical_effect/E, stage = 1)
+	if(has_medical_effect(E))
+		return
+	var/datum/medical_effect/ME = new E
+	ME.stage = stage
+	medical_effects += ME
+	return
+
+/mob/living/carbon/proc/remove_medical_effect(var/datum/medical_effect/E)
+	for(var/datum/medical_effect/ME in medical_effects)
+		if(istype(ME, E))
+			medical_effects -= ME
+	return
+
+/mob/living/carbon/proc/has_medical_effect(var/datum/medical_effect/E)
+	for(var/datum/medical_effect/MED in medical_effects)
+		if(istype(MED, E))
+			return 1
+
+/mob/living/carbon/handle_critical()
+	if(health <= config.health_threshold_crit)
+		add_medical_effect(/datum/medical_effect/shock)
+		if(health >= -51)
+			adjustOxyLoss(1)
+		if(health <= -100)
+			adjustOxyLoss(1)
+			if(prob(15))
+				add_medical_effect(/datum/medical_effect/cardiac_arrest, 1)
+			var/total_health = health - getBrainLoss()
+			var/kill_prob = total_health / -100
+			if(prob(kill_prob))
+				death()
+			return

```

Comment:
this is going to nerf xenos. Once they reach -100 and get cardiac arrest even xeno weeds won't be able to heal them faster than the oxyloss. Also, not all carbon currently die at -100 so it's a nerf for alien queen and a buff for alien larva.